### PR TITLE
[Enhancement] Support enable_show_all_variables flag && remove unused code

### PR DESCRIPTION
**Pull request by Seaven**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

1. Add enable_show_all_variables to show invisible variables
2. Remove deprecated variables

```
MySQL td> set enable_show_all_variables=true;
Query OK, 0 rows affected
Time: 0.002s
MySQL td> show variables;
+---------------------------------------------+---------------------------+
| Variable_name                               | Value                     |
+---------------------------------------------+---------------------------+
| ............                                                            |
| cbo_cte_reuse                               | true                      |
| cbo_cte_reuse_rate                          | 1.2                       |
| cbo_enable_dp_join_reorder                  | true                      |
| cbo_enable_greedy_join_reorder              | true                      |
| cbo_enable_low_cardinality_optimize         | true                      |
| cbo_max_reorder_node                        | 50                        |
| cbo_max_reorder_node_use_dp                 | 10                        |
| cbo_max_reorder_node_use_exhaustive         | 4                         |
| ............                                                            |
+---------------------------------------------+---------------------------+
97 rows in set
Time: 0.014s
MySQL td> set enable_show_all_variables=false;
Query OK, 0 rows affected
Time: 0.001s
MySQL td> show variables;
+--------------------------------------------+---------------------------+
| Variable_name                              | Value                     |
+--------------------------------------------+---------------------------+
| ................                                                       |
| broadcast_row_limit                        | 15000000                  |
| cbo_cte_reuse                              | true                      |
| cbo_enable_low_cardinality_optimize        | true                      |
| cbo_max_reorder_node_use_dp                | 10                        |
| cbo_max_reorder_node_use_exhaustive        | 4                         |
| character_set_client                       | utf8                      |
| character_set_connection                   | utf8                      |
| character_set_database                     | utf8                      |
| ................                                                       |
+--------------------------------------------+---------------------------+
80 rows in set
Time: 0.021s
MySQL td>
```

### CODE REVIEW

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 29 / 31 (93.55%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|23|25|92.00%|
|:large_blue_circle:|com/starrocks/qe/VariableMgr.java|6|6|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 1]
```diff
@@ -307,7 +318,6 @@ public class SessionVariable implements Serializable, Writable, Cloneable {
     private String collationConnection = "utf8_general_ci";
     @VariableMgr.VarAttr(name = COLLATION_DATABASE)
     private String collationDatabase = "utf8_general_ci";
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:320]: Do we really support collation?

**Review by murphyatwork — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java (hunk 1 above)
Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:320]: No, just for compatibility

**Review by Youngwb — APPROVED** [review]

**Pull request merged**